File: LeetCode/1446. Consecutive Characters.cpp
/*** KNOWLEDGE IS POWER  ***/

class Solution {
public:
    int maxPower(string s) {
        int n = s.length() ;
        if (n < 2) 
        {
            return n ;
        }
        char prev = ' ' ;
        int current = 0 , ans = 1 ;
        for(int i = 0; i < n; i++) 
        {
            char c = s[i] ;
            if (c == prev) 
            {
                current++;
                ans = max(current , ans) ;
            } 
            else
            {
                current = 1 ;
                prev = c ;
            }
        }
        return ans ;
    }
};

File: LeetCode/1446. Consecutive Characters.py
### KNOWLEDGE IS POWER  ###

class Solution(object):
    def maxPower(self, s):
        """
        :type s: str
        :rtype: int
        """
        longest = 0
        count = 0
        prev = None
        for i, c in enumerate(s):
            if c == prev:
                count += 1                
            else:
                longest = max(longest, count)
                prev = c          
                count = 1                
        return max(longest, count) # If iteration reaches end & else condition isn't satisfied
